pkg/tar/tar.go

```diff
@@ -38,10 +39,47 @@ var ErrNotSupportedPlatform = errors.New("platform and architecture is not suppo
 // root of the tar file and should be cleaned (for example using filepath.Clean)
 type PathWhitelistMap map[string]struct{}
 
-// extractTar extracts a tarball (from a tar.Reader) into the root directory
-// if pwl is not nil, only the paths in the map are extracted.
-// If overwrite is true, existing files will be overwritten.
-func extractTar(tr *tar.Reader, overwrite bool, pwl PathWhitelistMap, uidRange *uid.UidRange) error {
+type FilePermissionsEditor func(string, int, int, byte, os.FileInfo) error
+
+func NewUidShiftingFilePermEditor(uidRange *uid.UidRange) (FilePermissionsEditor, error) {
+	user, err := user.Current()
```

Comment:
Unless you actually need other user fields just use os.Geteuid()

On Mon, Oct 19, 2015 at 10:26 AM, Derek Gonyeo <notifications@github.com>
wrote:

> In pkg/tar/tar.go
> <https://github.com/coreos/rkt/pull/1616#discussion_r42399857>:
>
> > @@ -38,10 +39,47 @@ var ErrNotSupportedPlatform = errors.New("platform and architecture is not suppo
> >  // root of the tar file and should be cleaned (for example using filepath.Clean)
> >  type PathWhitelistMap map[string]struct{}
> >
> > -// extractTar extracts a tarball (from a tar.Reader) into the root directory
> > -// if pwl is not nil, only the paths in the map are extracted.
> > -// If overwrite is true, existing files will be overwritten.
> > -func extractTar(tr *tar.Reader, overwrite bool, pwl PathWhitelistMap, uidRange *uid.UidRange) error {
> > +type FilePermissionsEditor func(string, int, int, byte, os.FileInfo) error
> > +
> > +func NewUidShiftingFilePermEditor(uidRange *uid.UidRange) (FilePermissionsEditor, error) {
> > +	user, err := user.Current()
>
> Perhaps on err it could just assume it's running as root? I feel like I
> might be missing some edge case with that though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/rkt/pull/1616/files#r42399857>.
>